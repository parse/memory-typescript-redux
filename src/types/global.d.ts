declare module 'enzyme-adapter-react-16';
declare module 'redux-saga-test-plan';
