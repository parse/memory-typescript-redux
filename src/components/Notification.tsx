import * as React from 'react';
import { Severity, Notification as NotificationType } from '../types/index';

interface NotificationProps {
  notification: NotificationType;
}

export default class Notification extends React.Component<
  NotificationProps,
  {}
> {
  render() {
    const { notification } = this.props;

    const content = () => {
      if (notification.severity === Severity.Info) {
        return (
          <strong style={{ color: 'orange' }}>{notification.message}</strong>
        );
      } else if (notification.severity === Severity.Success) {
        return (
          <strong style={{ color: 'green' }}>{notification.message}</strong>
        );
      } else if (notification.severity === Severity.Error) {
        return <strong style={{ color: 'red' }}>{notification.message}</strong>;
      } else {
        return null;
      }
    };

    return <div className={'notificationCenter notification'}>{content()}</div>;
  }
}
