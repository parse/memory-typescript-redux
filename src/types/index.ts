export enum GameStatus {
  NotStarted = 0,
  Ongoing = 1,
  Ended = 2,
}

export enum Severity {
  Info,
  Success,
  Error,
}

export interface StoreState {
  readonly memory: MemoryState;
  readonly notification: NotificationState;
}

export interface MemoryState {
  readonly tiles: Array<Tile>;
  readonly isWaiting: Boolean;
  readonly numberOfTries: number;
  readonly status: GameStatus;
}

export interface NotificationState {
  readonly notifications: Array<Notification>;
}

export interface Tile {
  readonly image: String;
  readonly tileKey: number;
  readonly flipped: Boolean;
  readonly matched: Boolean;
}

export interface Notification {
  readonly uid: number;
  readonly message: String;
  readonly severity: Severity;
}
