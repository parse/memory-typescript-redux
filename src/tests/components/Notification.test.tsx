import * as React from 'react';
import Notification from '../../components/Notification';
import * as renderer from 'react-test-renderer';
import { Severity } from '../../types/index';

it('renders correctly', () => {
  const notification = {
    uid: 1,
    severity: Severity.Error,
    message: 'Lol',
  };

  const tree = renderer
    .create(<Notification notification={notification} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
