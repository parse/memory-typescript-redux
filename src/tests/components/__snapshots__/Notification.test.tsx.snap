// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="notificationCenter notification"
>
  <strong
    style={
      Object {
        "color": "red",
      }
    }
  >
    Lol
  </strong>
</div>
`;
